docs/elastic_inference.md

```diff
@@ -0,0 +1,182 @@
+    
+    def inference(self, model_input):
+        # NOT COMPLETE CODE, refer template above for it.
+        .....
+        .....
+        .....     
+        model_input = [item.as_in_context(self.data_ctx) for item in model_input]
+
+```
+
+The above code shows initialization of two contexts, one for data (on CPU) and other for symbols (on EIA).
+
+Once we have the code ready. We can build a model archive consumable by MMS, using the model-archiver. The [model-archiver](https://github.com/awslabs/mxnet-model-server/blob/master/model-archiver/README.md) tool enables the create to an archive understood by MMS.
+
+```python
+model-archiver --model-name <model-name> --handler model_service:handle --export-path <output-dir> --model-path <model_dir> --runtime python
+```
+
+This will create file ```<model-name>.mar``` in the directory ```<output-dir>```.
+
+## Running elastic inference on a resnet-152
+A prebuilt Amazon Elastic Inference supporting resnet-152  model archive can be downloaded using the following command:
+
+```bash
+$ wget https://s3.amazonaws.com/model-server/model_archive_1.0/resnet-152-eia.mar
+```
+**NOTE:** The above archive will exclusively only work with EIA enabled instances, of EC2 and DLAMI v20.
+
+Start the EC2 instance, with DLAMI v20There are two conda environments with mxnet elastic inference and mxnet-model-server pre-installed. Either of the two can be used.
+```bash
+# python 3 
+$ source activate amazonei_mxnet_p36
+
+# python 2 
+$ source activate amazonei_mxnet_p27
+```
+
+After entering the virtual environment. We start MMS.
+```bash
+# Start MMS
+$ mxnet-model-server --start
+```
+
+We register the Resnet-152 model and start a single worker for it
+
+```bash
+# Register resent-18 model archive with Amazon EI support
+$ curl -X POST 127.0.0.1:8081/models?url=https://s3.amazonaws.com/model-server/model_archive_1.0/resnet-152-eia.mar
+
+# Start a single worker for the registered model
+
+$ curl -X PUT '127.0.0.1:8081/models/resnet-152?min_worker=1&synchronous=true'
+
+```
+
+Now the model is ready for some inference requests. Let us download a kitten image for classification.
+```bash
+$ curl -O https://s3.amazonaws.com/model-server/inputs/kitten.jpg
+$ curl -X POST http://127.0.0.1:8080/predictions/resnet-152 -T kitten.jpg
+```
+
+The predict endpoint will return a prediction response in JSON. It will look something like the following result:
+
+```json
+[
+  {
+    "probability": 0.7148934602737427,
+    "class": "n02123045 tabby, tabby cat"
+  },
+  {
+    "probability": 0.22877734899520874,
+    "class": "n02123159 tiger cat"
+  },
+  {
+    "probability": 0.04032360762357712,
+    "class": "n02124075 Egyptian cat"
+  },
+  {
+    "probability": 0.008370809257030487,
+    "class": "n02127052 lynx, catamount"
+  },
+  {
+    "probability": 0.0006728142034262419,
+    "class": "n02129604 tiger, Panthera tigris"
+  }
+```
```

Comment:
This looks strange, they are all red

Reply:
I imagine that if get rid of the `text` code block, you might find things render better...